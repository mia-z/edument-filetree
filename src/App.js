import React, { useReducer } from 'react';
import './App.scss';
import "bootstrap/dist/css/bootstrap.min.css";
import { HashRouter as Router } from "react-router-dom";
import { Container } from "react-bootstrap";
import DirectoryView from "./Components/DirectoryView";
import FilePath from "./Components/SubComponents/FilePath";
import { DirectoryReducer, initialState } from "./Reducers/DirectoryReducer";
import BoxView from "./Components/BoxView";
import MenuBar from "./Components/MenuBar";

function App() {
    const [state, dispatch] = useReducer(DirectoryReducer, initialState);

    return (
        <Router>
            <Container>
                <BoxView>
                    <MenuBar DirectoryState={state} dispatch={dispatch} />
                    <FilePath DirectoryState={state} dispatch={dispatch} />
                    <DirectoryView DirectoryState={state} dispatch={dispatch} />
                </BoxView>
            </Container>
        </Router>
    );
}

export default App;
