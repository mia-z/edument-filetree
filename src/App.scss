@import "scss/vars";

html, div#root {
    background-color: $secondhand-grey;
}
