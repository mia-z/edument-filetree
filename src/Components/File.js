import React from 'react';

export const File = ({name}) => {
    return(
        <div>
            {name}
        </div>
    );
}

export default File;
