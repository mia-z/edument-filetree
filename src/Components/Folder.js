import React from 'react';
import { Link } from "react-router-dom";

export const Folder = ({name, handleFolderClick}) => {

    return(
        <div onClick={() => handleFolderClick(name)}>
            {name}
        </div>
    );
}

export default Folder;
