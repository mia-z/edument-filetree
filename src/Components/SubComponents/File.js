import React, { useState } from 'react';
import OptionsButton from "./OptionsButton";
import ContextualOverlay from "./ContextualOverlay";

export const File = ({file, DispatchContext}) => {
    //The following two variables only assumes that the filename ONLY contains a single (1) period, splitting the name from the extension.
    const filename = file.Name.split(".")[0];
    const fileExtension = file.Name.split(".")[1];

    const [selected, setSelected] = useState(false);

    const getExtensionImage = (extension) => {
        switch (extension) {
            case "txt": return "fa-file-alt";
            case "png":
            case "jpg":
            case "jpeg":
            case "bmp": return "fa-file-image";
            case "zip":
            case "rar":
            case "7z":
            case "gzip": return "fa-file-archive";
            default: return "fa-file";
        }
    }

    const handleOptionsClick = () => setSelected(!selected);

    return(
        <div className={"directory-item"}>
            <OptionsButton click={() => handleOptionsClick()} isActive={selected}/>
            <ContextualOverlay selected={selected} objectProps={file} hideContextual={() => setSelected(false)} DispatchContext={DispatchContext}/>
            <i className={`mt-3 far ${getExtensionImage(fileExtension)} fa-5x`} />
            <div className={"trunc-container"}>
                <div className={"file-name"}>
                    {filename}.
                </div>
                <div className={"file-extension"}>
                    {fileExtension}
                </div>
            </div>
        </div>
    );
}

export default File;
