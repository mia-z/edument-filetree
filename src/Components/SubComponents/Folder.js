import React from 'react';

export const Folder = ({name, handleFolderClick}) => {

    return(
        <div onClick={() => handleFolderClick(name)}>
            {name}
        </div>
    );
}

export default Folder;
