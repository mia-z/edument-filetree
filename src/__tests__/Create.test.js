import React from 'react';
import App from '../App';
import { render, fireEvent, screen } from "@testing-library/react";
import Folder from "../Components/SubComponents/Folder";
import File from "../Components/SubComponents/File";

describe("Tests to make sure the create functions work", () => {
    it("Make sure the create file button renders into view", () => {
        render(<App />)
        expect(screen.getByText(/Create new file/)).toBeVisible();
    });

    it("Make sure the correct modal appears when trying to create a new file", () => {
        let app = render(<App />)
        let button = screen.getByText(/Create new file/);
        fireEvent.click(button);
        expect(app).toMatchSnapshot();
    });

    it("Make sure a new file is created when modal is submitted", () => {
        let app = render(<App />)
        fireEvent.click(screen.getByText(/Create new file/));
        fireEvent.change(screen.getByPlaceholderText(/Enter new name../), { target: { value: "Marvelsucks.txt" } });
        fireEvent.click(screen.getByText(/Confirm/));
        expect(screen.getByText(/Marvelsucks./)).toBeVisible();
    });

    it("Make sure the create folder button renders into view", () => {
        render(<App />)
        expect(screen.getByText(/Create new folder/)).toBeVisible();
    });

    it("Make sure the correct modal appears when trying to create a new folder", () => {
        let app = render(<App />)
        let button = screen.getByText(/Create new folder/);
        fireEvent.click(button);
        expect(app).toMatchSnapshot();
    });

    it("Make sure a new folder is created when modal is submitted", () => {
        let app = render(<App />)
        fireEvent.click(screen.getByText(/Create new folder/));
        fireEvent.change(screen.getByPlaceholderText(/Enter new name../), { target: { value: "dcuniverse" } });
        fireEvent.click(screen.getByText(/Confirm/));
        expect(screen.getByText(/dcuniverse/)).toBeVisible();
    });
});

