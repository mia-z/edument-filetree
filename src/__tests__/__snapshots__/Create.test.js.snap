// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests to make sure the create functions work Make sure the correct modal appears when trying to create a new file 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="modal-open"
    style="padding-right: 0px;"
  >
    <div
      aria-hidden="true"
    >
      <div
        class="container"
      >
        <div
          class="box"
        >
          <div
            class="container"
          >
            <div
              class="menu-bar"
            >
              <button>
                Create new file
              </button>
              <button>
                Create new folder
              </button>
            </div>
          </div>
          <div
            class="path-container container-fluid"
          >
            <div
              class="align-items-center mx-auto row no-gutters"
            >
              <div
                class="col-md-1"
              >
                <button
                  class="btn btn-secondary btn-block"
                  disabled=""
                  id="back-button"
                  type="button"
                >
                  <i
                    class="fas fa-level-up-alt"
                  />
                </button>
              </div>
              <div
                class="col-md-11"
              >
                <div
                  class="path-text-container"
                >
                  <div
                    class="path-link active"
                  >
                    /
                    root
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="directory-container"
          >
            <div
              class="directory-item"
            >
              <div
                class="options-button"
              >
                <i
                  class="fas fa-ellipsis-h fa-fw"
                />
              </div>
              <div
                class="contextual-overlay hidden"
              >
                <div
                  class="context-button hidden"
                >
                  <div
                    class="context-button-text hidden"
                  >
                    Rename
                  </div>
                </div>
                <div
                  class="context-button hidden"
                >
                  <div
                    class="context-button-text hidden"
                  >
                    Delete
                  </div>
                </div>
              </div>
              <i
                class="mt-3 far fa-file-alt fa-5x"
              />
              <div
                class="trunc-container"
              >
                <div
                  class="file-name"
                >
                  fact_marvel_beats_dc
                  .
                </div>
                <div
                  class="file-extension"
                >
                  txt
                </div>
              </div>
            </div>
            <div
              class="directory-item folder"
            >
              <div
                class="options-button"
              >
                <i
                  class="fas fa-ellipsis-h fa-fw"
                />
              </div>
              <div
                class="contextual-overlay hidden"
              >
                <div
                  class="context-button hidden"
                >
                  <div
                    class="context-button-text hidden"
                  >
                    Rename
                  </div>
                </div>
                <div
                  class="context-button hidden"
                >
                  <div
                    class="context-button-text hidden"
                  >
                    Delete
                  </div>
                </div>
              </div>
              <i
                class="mt-4 far fa-folder fa-5x"
              />
              <div
                class="trunc-container"
              >
                <div
                  class="file-name"
                >
                  marvel
                </div>
              </div>
            </div>
            <div
              class="directory-item folder"
            >
              <div
                class="options-button"
              >
                <i
                  class="fas fa-ellipsis-h fa-fw"
                />
              </div>
              <div
                class="contextual-overlay hidden"
              >
                <div
                  class="context-button hidden"
                >
                  <div
                    class="context-button-text hidden"
                  >
                    Rename
                  </div>
                </div>
                <div
                  class="context-button hidden"
                >
                  <div
                    class="context-button-text hidden"
                  >
                    Delete
                  </div>
                </div>
              </div>
              <i
                class="mt-4 far fa-folder fa-5x"
              />
              <div
                class="trunc-container"
              >
                <div
                  class="file-name"
                >
                  dc
                </div>
              </div>
            </div>
            <div
              class="directory-item folder"
            >
              <div
                class="options-button"
              >
                <i
                  class="fas fa-ellipsis-h fa-fw"
                />
              </div>
              <div
                class="contextual-overlay hidden"
              >
                <div
                  class="context-button hidden"
                >
                  <div
                    class="context-button-text hidden"
                  >
                    Rename
                  </div>
                </div>
                <div
                  class="context-button hidden"
                >
                  <div
                    class="context-button-text hidden"
                  >
                    Delete
                  </div>
                </div>
              </div>
              <i
                class="mt-4 far fa-folder fa-5x"
              />
              <div
                class="trunc-container"
              >
                <div
                  class="file-name"
                >
                  very
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="fade modal-backdrop show"
    />
    <div
      aria-modal="true"
      class="fade modal show"
      role="dialog"
      style="display: block;"
      tabindex="-1"
    >
      <div
        class="modal-dialog modal-lg modal-dialog-centered"
        role="document"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <div
              class="container"
            >
              <div
                class="justify-content-center row"
              >
                <div
                  class="col-md-6"
                >
                  <h4
                    class="text-center text-capitalize"
                  >
                    Create 
                    file
                  </h4>
                </div>
              </div>
            </div>
          </div>
          <div
            class="modal-body"
          >
            <div
              class="container"
            >
              <div
                class="row"
              >
                <div
                  class="col"
                >
                  <div
                    class="fade alert alert-primary show"
                    role="alert"
                  >
                    <p
                      class="text-center my-auto"
                    >
                      <b>
                        Creating 
                        file
                         in: 
                      </b>
                      root
                    </p>
                  </div>
                </div>
              </div>
              <div
                class="justify-content-center mb-4 row"
              >
                <div
                  class="col-md-12"
                >
                  <input
                    class="is-invalid is-valid form-control"
                    placeholder="Enter new name.."
                    value=""
                  />
                  <small
                    class="form-text text-muted"
                  >
                    Filenames may only contain letters, numbers dashes and underscores, and must have a file extension
                  </small>
                </div>
              </div>
              <div
                class="justify-content-center row"
              >
                <div
                  class="col-md-4"
                >
                  <button
                    class="btn btn-danger btn-block"
                    type="button"
                  >
                    Cancel
                  </button>
                </div>
                <div
                  class="col-md-4"
                >
                  <button
                    class="btn btn-primary btn-block"
                    disabled=""
                    type="button"
                  >
                    Confirm
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  >
    <div
      class="container"
    >
      <div
        class="box"
      >
        <div
          class="container"
        >
          <div
            class="menu-bar"
          >
            <button>
              Create new file
            </button>
            <button>
              Create new folder
            </button>
          </div>
        </div>
        <div
          class="path-container container-fluid"
        >
          <div
            class="align-items-center mx-auto row no-gutters"
          >
            <div
              class="col-md-1"
            >
              <button
                class="btn btn-secondary btn-block"
                disabled=""
                id="back-button"
                type="button"
              >
                <i
                  class="fas fa-level-up-alt"
                />
              </button>
            </div>
            <div
              class="col-md-11"
            >
              <div
                class="path-text-container"
              >
                <div
                  class="path-link active"
                >
                  /
                  root
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="directory-container"
        >
          <div
            class="directory-item"
          >
            <div
              class="options-button"
            >
              <i
                class="fas fa-ellipsis-h fa-fw"
              />
            </div>
            <div
              class="contextual-overlay hidden"
            >
              <div
                class="context-button hidden"
              >
                <div
                  class="context-button-text hidden"
                >
                  Rename
                </div>
              </div>
              <div
                class="context-button hidden"
              >
                <div
                  class="context-button-text hidden"
                >
                  Delete
                </div>
              </div>
            </div>
            <i
              class="mt-3 far fa-file-alt fa-5x"
            />
            <div
              class="trunc-container"
            >
              <div
                class="file-name"
              >
                fact_marvel_beats_dc
                .
              </div>
              <div
                class="file-extension"
              >
                txt
              </div>
            </div>
          </div>
          <div
            class="directory-item folder"
          >
            <div
              class="options-button"
            >
              <i
                class="fas fa-ellipsis-h fa-fw"
              />
            </div>
            <div
              class="contextual-overlay hidden"
            >
              <div
                class="context-button hidden"
              >
                <div
                  class="context-button-text hidden"
                >
                  Rename
                </div>
              </div>
              <div
                class="context-button hidden"
              >
                <div
                  class="context-button-text hidden"
                >
                  Delete
                </div>
              </div>
            </div>
            <i
              class="mt-4 far fa-folder fa-5x"
            />
            <div
              class="trunc-container"
            >
              <div
                class="file-name"
              >
                marvel
              </div>
            </div>
          </div>
          <div
            class="directory-item folder"
          >
            <div
              class="options-button"
            >
              <i
                class="fas fa-ellipsis-h fa-fw"
              />
            </div>
            <div
              class="contextual-overlay hidden"
            >
              <div
                class="context-button hidden"
              >
                <div
                  class="context-button-text hidden"
                >
                  Rename
                </div>
              </div>
              <div
                class="context-button hidden"
              >
                <div
                  class="context-button-text hidden"
                >
                  Delete
                </div>
              </div>
            </div>
            <i
              class="mt-4 far fa-folder fa-5x"
            />
            <div
              class="trunc-container"
            >
              <div
                class="file-name"
              >
                dc
              </div>
            </div>
          </div>
          <div
            class="directory-item folder"
          >
            <div
              class="options-button"
            >
              <i
                class="fas fa-ellipsis-h fa-fw"
              />
            </div>
            <div
              class="contextual-overlay hidden"
            >
              <div
                class="context-button hidden"
              >
                <div
                  class="context-button-text hidden"
                >
                  Rename
                </div>
              </div>
              <div
                class="context-button hidden"
              >
                <div
                  class="context-button-text hidden"
                >
                  Delete
                </div>
              </div>
            </div>
            <i
              class="mt-4 far fa-folder fa-5x"
            />
            <div
              class="trunc-container"
            >
              <div
                class="file-name"
              >
                very
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Tests to make sure the create functions work Make sure the correct modal appears when trying to create a new folder 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="modal-open"
    style="padding-right: 0px;"
  >
    <div
      aria-hidden="true"
    >
      <div
        class="container"
      >
        <div
          class="box"
        >
          <div
            class="container"
          >
            <div
              class="menu-bar"
            >
              <button>
                Create new file
              </button>
              <button>
                Create new folder
              </button>
            </div>
          </div>
          <div
            class="path-container container-fluid"
          >
            <div
              class="align-items-center mx-auto row no-gutters"
            >
              <div
                class="col-md-1"
              >
                <button
                  class="btn btn-secondary btn-block"
                  disabled=""
                  id="back-button"
                  type="button"
                >
                  <i
                    class="fas fa-level-up-alt"
                  />
                </button>
              </div>
              <div
                class="col-md-11"
              >
                <div
                  class="path-text-container"
                >
                  <div
                    class="path-link active"
                  >
                    /
                    root
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="directory-container"
          >
            <div
              class="directory-item"
            >
              <div
                class="options-button"
              >
                <i
                  class="fas fa-ellipsis-h fa-fw"
                />
              </div>
              <div
                class="contextual-overlay hidden"
              >
                <div
                  class="context-button hidden"
                >
                  <div
                    class="context-button-text hidden"
                  >
                    Rename
                  </div>
                </div>
                <div
                  class="context-button hidden"
                >
                  <div
                    class="context-button-text hidden"
                  >
                    Delete
                  </div>
                </div>
              </div>
              <i
                class="mt-3 far fa-file-alt fa-5x"
              />
              <div
                class="trunc-container"
              >
                <div
                  class="file-name"
                >
                  fact_marvel_beats_dc
                  .
                </div>
                <div
                  class="file-extension"
                >
                  txt
                </div>
              </div>
            </div>
            <div
              class="directory-item folder"
            >
              <div
                class="options-button"
              >
                <i
                  class="fas fa-ellipsis-h fa-fw"
                />
              </div>
              <div
                class="contextual-overlay hidden"
              >
                <div
                  class="context-button hidden"
                >
                  <div
                    class="context-button-text hidden"
                  >
                    Rename
                  </div>
                </div>
                <div
                  class="context-button hidden"
                >
                  <div
                    class="context-button-text hidden"
                  >
                    Delete
                  </div>
                </div>
              </div>
              <i
                class="mt-4 far fa-folder fa-5x"
              />
              <div
                class="trunc-container"
              >
                <div
                  class="file-name"
                >
                  marvel
                </div>
              </div>
            </div>
            <div
              class="directory-item folder"
            >
              <div
                class="options-button"
              >
                <i
                  class="fas fa-ellipsis-h fa-fw"
                />
              </div>
              <div
                class="contextual-overlay hidden"
              >
                <div
                  class="context-button hidden"
                >
                  <div
                    class="context-button-text hidden"
                  >
                    Rename
                  </div>
                </div>
                <div
                  class="context-button hidden"
                >
                  <div
                    class="context-button-text hidden"
                  >
                    Delete
                  </div>
                </div>
              </div>
              <i
                class="mt-4 far fa-folder fa-5x"
              />
              <div
                class="trunc-container"
              >
                <div
                  class="file-name"
                >
                  dc
                </div>
              </div>
            </div>
            <div
              class="directory-item folder"
            >
              <div
                class="options-button"
              >
                <i
                  class="fas fa-ellipsis-h fa-fw"
                />
              </div>
              <div
                class="contextual-overlay hidden"
              >
                <div
                  class="context-button hidden"
                >
                  <div
                    class="context-button-text hidden"
                  >
                    Rename
                  </div>
                </div>
                <div
                  class="context-button hidden"
                >
                  <div
                    class="context-button-text hidden"
                  >
                    Delete
                  </div>
                </div>
              </div>
              <i
                class="mt-4 far fa-folder fa-5x"
              />
              <div
                class="trunc-container"
              >
                <div
                  class="file-name"
                >
                  very
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="fade modal-backdrop show"
    />
    <div
      aria-modal="true"
      class="fade modal show"
      role="dialog"
      style="display: block;"
      tabindex="-1"
    >
      <div
        class="modal-dialog modal-lg modal-dialog-centered"
        role="document"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <div
              class="container"
            >
              <div
                class="justify-content-center row"
              >
                <div
                  class="col-md-6"
                >
                  <h4
                    class="text-center text-capitalize"
                  >
                    Create 
                    folder
                  </h4>
                </div>
              </div>
            </div>
          </div>
          <div
            class="modal-body"
          >
            <div
              class="container"
            >
              <div
                class="row"
              >
                <div
                  class="col"
                >
                  <div
                    class="fade alert alert-primary show"
                    role="alert"
                  >
                    <p
                      class="text-center my-auto"
                    >
                      <b>
                        Creating 
                        folder
                         in: 
                      </b>
                      root
                    </p>
                  </div>
                </div>
              </div>
              <div
                class="justify-content-center mb-4 row"
              >
                <div
                  class="col-md-12"
                >
                  <input
                    class="is-invalid is-valid form-control"
                    placeholder="Enter new name.."
                    value=""
                  />
                  <small
                    class="form-text text-muted"
                  >
                    Folder names may only contains letters, numbers, dashes and underscores
                  </small>
                </div>
              </div>
              <div
                class="justify-content-center row"
              >
                <div
                  class="col-md-4"
                >
                  <button
                    class="btn btn-danger btn-block"
                    type="button"
                  >
                    Cancel
                  </button>
                </div>
                <div
                  class="col-md-4"
                >
                  <button
                    class="btn btn-primary btn-block"
                    disabled=""
                    type="button"
                  >
                    Confirm
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  >
    <div
      class="container"
    >
      <div
        class="box"
      >
        <div
          class="container"
        >
          <div
            class="menu-bar"
          >
            <button>
              Create new file
            </button>
            <button>
              Create new folder
            </button>
          </div>
        </div>
        <div
          class="path-container container-fluid"
        >
          <div
            class="align-items-center mx-auto row no-gutters"
          >
            <div
              class="col-md-1"
            >
              <button
                class="btn btn-secondary btn-block"
                disabled=""
                id="back-button"
                type="button"
              >
                <i
                  class="fas fa-level-up-alt"
                />
              </button>
            </div>
            <div
              class="col-md-11"
            >
              <div
                class="path-text-container"
              >
                <div
                  class="path-link active"
                >
                  /
                  root
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="directory-container"
        >
          <div
            class="directory-item"
          >
            <div
              class="options-button"
            >
              <i
                class="fas fa-ellipsis-h fa-fw"
              />
            </div>
            <div
              class="contextual-overlay hidden"
            >
              <div
                class="context-button hidden"
              >
                <div
                  class="context-button-text hidden"
                >
                  Rename
                </div>
              </div>
              <div
                class="context-button hidden"
              >
                <div
                  class="context-button-text hidden"
                >
                  Delete
                </div>
              </div>
            </div>
            <i
              class="mt-3 far fa-file-alt fa-5x"
            />
            <div
              class="trunc-container"
            >
              <div
                class="file-name"
              >
                fact_marvel_beats_dc
                .
              </div>
              <div
                class="file-extension"
              >
                txt
              </div>
            </div>
          </div>
          <div
            class="directory-item folder"
          >
            <div
              class="options-button"
            >
              <i
                class="fas fa-ellipsis-h fa-fw"
              />
            </div>
            <div
              class="contextual-overlay hidden"
            >
              <div
                class="context-button hidden"
              >
                <div
                  class="context-button-text hidden"
                >
                  Rename
                </div>
              </div>
              <div
                class="context-button hidden"
              >
                <div
                  class="context-button-text hidden"
                >
                  Delete
                </div>
              </div>
            </div>
            <i
              class="mt-4 far fa-folder fa-5x"
            />
            <div
              class="trunc-container"
            >
              <div
                class="file-name"
              >
                marvel
              </div>
            </div>
          </div>
          <div
            class="directory-item folder"
          >
            <div
              class="options-button"
            >
              <i
                class="fas fa-ellipsis-h fa-fw"
              />
            </div>
            <div
              class="contextual-overlay hidden"
            >
              <div
                class="context-button hidden"
              >
                <div
                  class="context-button-text hidden"
                >
                  Rename
                </div>
              </div>
              <div
                class="context-button hidden"
              >
                <div
                  class="context-button-text hidden"
                >
                  Delete
                </div>
              </div>
            </div>
            <i
              class="mt-4 far fa-folder fa-5x"
            />
            <div
              class="trunc-container"
            >
              <div
                class="file-name"
              >
                dc
              </div>
            </div>
          </div>
          <div
            class="directory-item folder"
          >
            <div
              class="options-button"
            >
              <i
                class="fas fa-ellipsis-h fa-fw"
              />
            </div>
            <div
              class="contextual-overlay hidden"
            >
              <div
                class="context-button hidden"
              >
                <div
                  class="context-button-text hidden"
                >
                  Rename
                </div>
              </div>
              <div
                class="context-button hidden"
              >
                <div
                  class="context-button-text hidden"
                >
                  Delete
                </div>
              </div>
            </div>
            <i
              class="mt-4 far fa-folder fa-5x"
            />
            <div
              class="trunc-container"
            >
              <div
                class="file-name"
              >
                very
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
